src/tree.rs

```diff
@@ -705,10 +731,11 @@ impl Node {
   /// Creates a new `Node` as child to the self `Node`
   pub fn add_child(&mut self, child: Node) -> Result<Node, ()> {
     unsafe {
-      if xmlAddChild(self.node_ptr, child.node_ptr).is_null() {
+      let new_child_ptr = xmlAddChild(self.node_ptr, child.node_ptr);
+      if new_child_ptr.is_null() {
         Err(())
       } else {
-        Ok(child)
+        Ok(Node { node_ptr: new_child_ptr })
```

Comment:
this was actually a bug, as there is subtle merging going on when adding e.g. several adjacent text nodes